cadasta/gui/tools/wizard/project_creation_wizard.py

```diff
@@ -0,0 +1,209 @@
+        QDialog.__init__(self, parent)
+        self.setupUi(self)
+        self.setWindowTitle('Cadasta')
+        self.lblSubtitle.setText(
+            tr('Cadasta project creation wizard')
+        )
+
+        self.iface = iface
+        self.organisations_list = None
+        self.layer = None
+        self.parent_step = None
+        self.step_index = 1
+        self.step_length = 3
+        self.set_step_label()
+        self.message_bar = None
+
+        self.step_project_creation01 = StepProjectCreation1(self)
+        self.step_project_creation02 = StepProjectCreation2(self)
+        self.step_project_creation03 = StepProjectCreation3(self)
+
+        self.stackedWidget.addWidget(self.step_project_creation01)
+        self.stackedWidget.addWidget(self.step_project_creation02)
+        self.stackedWidget.addWidget(self.step_project_creation03)
+
+        self.steps = []
+
+        step = self.step_project_creation01
+        step.set_widgets()
+        self.go_to_step(step)
+        self.set_logo()
+
+    def set_logo(self):
+        filename = resources_path("images/white_icon.png")
+        LOGGER.debug(filename)
+        pixmap = QPixmap(filename)
+        self.lblMainIcon.setPixmap(pixmap)
+
+    def set_step_label(self):
+        """Display step label."""
+        self.lblStep.setText('%d/%d' % (self.step_index, self.step_length))
+
+    # ===========================
+    # NAVIGATION
+    # ===========================
+
+    def go_to_step(self, step):
+        """Set the stacked widget to the given step, set up the buttons,
+           and run all operations that should start immediately after
+           entering the new step.
+
+        :param step: The step widget to be moved to.
+        :type step: QWidget
+        """
+        self.stackedWidget.setCurrentWidget(step)
+
+        # Enable the Back button unless it's not the first step
+        self.pbnBack.setEnabled(
+            step not in [self.step_project_creation01] or
+            self.parent_step is not None)
+
+        # Set Next button label
+        if step == self.step_project_creation03:
+            self.pbnNext.setText(self.tr('Close'))
+        else:
+            self.pbnNext.setText(self.tr('Next'))
+
+    def get_current_step(self):
+        """Return current step of the wizard.
+
+        :returns: Current step of the wizard.
+        :rtype: WizardStep instance
+        """
+        return self.stackedWidget.currentWidget()
+
+    # prevents actions being handled twice
+    # noinspection PyPep8Naming
+    @pyqtSignature('')
+    def on_pbnNext_released(self):
+        """Handle the Next button release.
+
+        .. note:: This is an automatic Qt slot
+           executed when the Next button is released.
+        """
+        current_step = self.get_current_step()
+        valid_status, message = current_step.validate_step()
+
+        if not valid_status:
+            self.message_bar = QgsMessageBar()
+            self.message_bar.pushWarning(
```

Comment:
> Perhaps, since this is a wizard, it may cover the QGIS, and the QgsMessageBar is hidden from the user.

I was talking about the message bar in the form qwidget, not in the qgis interface.
I also like the UX from web.